charts/judge-web/templates/ingress.yaml

```diff
@@ -0,0 +1,122 @@
+{{- if .Values.ingress.enabled -}}
+{{- $fullName := include "judge-web.fullname" . -}}
+{{- $svcPort := .Values.service.port -}}
+{{- if and .Values.ingress.className (not (semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion)) }}
+  {{- if not (hasKey .Values.ingress.annotations "kubernetes.io/ingress.class") }}
+  {{- $_ := set .Values.ingress.annotations "kubernetes.io/ingress.class" .Values.ingress.className}}
+  {{- end }}
+{{- end }}
+{{- if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1
+{{- else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
```

Comment:
I appreciate these checks but, we should not be supporting anything less than k8s 1.27. Current version is 1.29 with 1.30 in progress. We should make sure supported versions is documented.